packages/core/src/operators/use/use.operator.ts

```diff
@@ -3,9 +3,9 @@ import { switchMap } from 'rxjs/operators';
 import { Middleware } from '../../effects/effects.interface';
 import { HttpRequest, HttpResponse } from '../../http.interface';
 
-export const use = <T, U, V>
-  (middleware: Middleware<T, U, V>, res?: HttpResponse) =>
-  (source$: Observable<HttpRequest>): Observable<HttpRequest<T, U, V>> =>
+export const use = <T extends HttpRequest>
```

Comment:
I think that input type is also needed for the middleware if the middleware is supposed to extend the type, i.e.
```
Req ->
  authMiddleware ->
(Req - from previous) & Tokens ->
  userMiddleware ->
(Req & Tokens - from previous) & UserData
```